packages/core/src/memory/memory.ts

```diff
@@ -247,13 +247,21 @@ export abstract class MastraMemory extends MastraBase {
    * @param messages - Array of messages to save
    * @returns Promise resolving to the saved messages
    */
-  abstract saveMessages({
-    messages,
-    memoryConfig,
-  }: {
-    messages: (MastraMessageV1 | MastraMessageV2)[];
-    memoryConfig: MemoryConfig | undefined;
+  abstract saveMessages(args: {
+    messages: (MastraMessageV1 | MastraMessageV2)[] | MastraMessageV1[] | MastraMessageV2[];
+    memoryConfig?: MemoryConfig | undefined;
+    format?: 'v1';
+  }): Promise<MastraMessageV1[]>;
+  abstract saveMessages(args: {
+    messages: (MastraMessageV1 | MastraMessageV2)[] | MastraMessageV1[] | MastraMessageV2[];
+    memoryConfig?: MemoryConfig | undefined;
+    format: 'v2';
   }): Promise<MastraMessageV2[]>;
+  abstract saveMessages(args: {
+    messages: (MastraMessageV1 | MastraMessageV2)[] | MastraMessageV1[] | MastraMessageV2[];
+    memoryConfig?: MemoryConfig | undefined;
+    format?: 'v1' | 'v2';
+  }): Promise<MastraMessageV2[] | MastraMessageV1[]>;
```

Comment:
by default it'll return the old format now, adding `format: "v2"` makes it return the new one. Whichever format is input/output the new format is stored in the DB still